chaco/data_view.py

```diff
@@ -242,7 +242,11 @@ def map_screen(self, data_array):
         """Maps an array of data points to screen space and returns an array
         of screen space points.
         """
-        # data_array is Nx2 array
+
+        # ensure data_array is an Nx2 ndarray
+        data_array = array(data_array)
+        data_array = data_array.reshape(-1,2)
```

Comment:
these are actually redundant / unneeded as in the return statement we call `array`